File: leetcode_75/Sliding_Window/python/1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}

        count = 0
        for i in range(k):
            if s[i] in vowels:
                count += 1
        mx = count

        for i in range(k, len(s)):
            if s[i - k] in vowels:
                count -= 1
            if s[i] in vowels:
                count += 1
            mx = max(mx, count)

        return mx

File: leetcode_75/Sliding_Window/cpp/1456.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int maxVowels(string s, int k)
    {
        string vowels = "aeiou";

        int count = 0;
        for (int i = 0; i < k; ++i)
        {
            if (vowels.find(s[i]) != string::npos)
            {
                count++;
            }
        }
        int mx = count;

        for (int i = k; i < s.size(); ++i)
        {
            if (vowels.find(s[i - k]) != string::npos)
            {
                count--;
            }
            if (vowels.find(s[i]) != string::npos)
            {
                count++;
            }
            mx = max(mx, count);
        }

        return mx;
    }
};